Warnings in building mimetype on GCC8

<!-- Before filling this issue, please read the wiki (https://github.com/fossology/fossology/wiki)
and search if the bug do not already exists in the issues (https://github.com/fossology/fossology/issues). -->

### Logs

```lisp
mimetype.c: In function ‘main’:
mimetype.c:226:11: warning: ‘strncpy’ specified bound 1024 equals destination size [-Wstringop-truncation]
           strncpy(A, PQgetvalue(result, i, 1), sizeof(A));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mimetype.c:162:5: warning: ‘strncpy’ specified bound 1024 equals destination size [-Wstringop-truncation]
     strncpy(A,argv[arg],sizeof(A));
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/mimetype/agent/mimetype.c
+++ b/src/mimetype/agent/mimetype.c
@@ -158,8 +158,8 @@ int main(int argc, char *argv[])
   for(arg=optind; arg < argc; arg++)
   {
     Akey = -1;
-    memset(A,'\0',sizeof(A));
-    strncpy(A,argv[arg],sizeof(A));
+    strncpy(A,argv[arg],sizeof(A)-1);
+    A[sizeof(A)-1] = '\0';
     DBCheckMime(A);
   }
 
@@ -223,7 +223,8 @@ int main(int argc, char *argv[])
         for(i=0; i < pfile_count; i++)
         {
           Akey = atoi(PQgetvalue(result, i, 0));
-          strncpy(A, PQgetvalue(result, i, 1), sizeof(A));
+          strncpy(A, PQgetvalue(result, i, 1), sizeof(A)-1);
+          A[sizeof(A)-1] = '\0';
           if (Akey <= 0 || A[0]=='\0')
           {
             printf("ERROR: Data is in an unknown format.\n");

